### PR TITLE
feat: add obs node server code

### PR DESCRIPTION
**Pull request by deer-hang**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
add obs nodeserver code to master
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/obs/nodeserver.go [hunk 1]
```diff
@@ -17,38 +17,147 @@ limitations under the License.
...
+
 func (ns *nodeServer) NodeStageVolume(_ context.Context, req *csi.NodeStageVolumeRequest) (
 	*csi.NodeStageVolumeResponse, error) {
+	log.Infof("NodeStageVolume: called with args %v", protosanitizer.StripSecrets(*req))
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
 func (ns *nodeServer) NodeUnstageVolume(_ context.Context, req *csi.NodeUnstageVolumeRequest) (
 	*csi.NodeUnstageVolumeResponse, error) {
+	log.Infof("NodeStageVolume: called with args %v", protosanitizer.StripSecrets(*req))
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:58]:
```suggestion
	log.Infof("NodeUnstageVolume: called with args %v", protosanitizer.StripSecrets(*req))
```

File: pkg/obs/nodeserver.go [hunk 2]
```diff
@@ -17,38 +17,147 @@ limitations under the License.
...
+		return &csi.NodePublishVolumeResponse{}, nil
+	}
+	if err := ns.Mount.MakeDir(targetPath); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to make dir: %s, error: %v", targetPath, err)
+	}
+	if err := createCredentialFile(ns.Driver.cloud.Global.AccessKey, ns.Driver.cloud.Global.SecretKey); err != nil {
+		return nil, err
+	}
+
+	mntCmd := fmt.Sprintf("obsfs %s %s -o url=obs.%s.%s -o passwd_file=%s %s", volume.BucketName, targetPath,
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:94]: Passing commands is not a good way, please refactor later.

File: pkg/obs/nodeserver.go [hunk 3]
```diff
@@ -17,38 +17,147 @@ limitations under the License.
...
+	log.Infof("NodePublishVolume: mount %s at %s successfully", volume.BucketName, targetPath)
+	return &csi.NodePublishVolumeResponse{}, nil
+}
+
+func createCredentialFile(accessKey, secretKey string) error {
+	if err := os.MkdirAll(path.Dir(credentialFile), os.ModePerm); err != nil {
+		return status.Errorf(codes.Internal, "Failed to make dir: %s, error: %v", credentialFile, err)
+	}
+
+	writer, err := os.OpenFile(filepath.Clean(credentialFile), os.O_CREATE|os.O_WRONLY|os.O_TRUNC, perm)
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:111]: Don't flush every time, this will cause errors in threads that are in use.

File: pkg/obs/nodeserver.go [hunk 4]
```diff
@@ -17,38 +17,147 @@ limitations under the License.
...
+	return nil
 }
 
 func (ns *nodeServer) NodeUnpublishVolume(_ context.Context, req *csi.NodeUnpublishVolumeRequest) (
 	*csi.NodeUnpublishVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	log.Infof("NodeUnpublishVolume: called with args %v", protosanitizer.StripSecrets(*req))
+	volumeID := req.GetVolumeId()
+	targetPath := req.GetTargetPath()
+	if len(volumeID) == 0 {
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:140]: It seems that the non-null check on the `volumeID`, and it is not called by other ways, such as calling the API. 
suggest to call the cloud API to check whether the resource exists.

File: pkg/obs/nodeserver.go [hunk 5]
```diff
@@ -17,38 +17,147 @@ limitations under the License.
...
+	}
+	if err := createCredentialFile(ns.Driver.cloud.Global.AccessKey, ns.Driver.cloud.Global.SecretKey); err != nil {
+		return nil, err
+	}
+
+	mntCmd := fmt.Sprintf("obsfs %s %s -o url=obs.%s.%s -o passwd_file=%s %s", volume.BucketName, targetPath,
+		ns.Driver.cloud.Global.Region, ns.Driver.cloud.Global.Cloud, credential, defaultOpts)
+	log.Infof("NodePublishVolume: obsfs cmd : %s", mntCmd)
+
+	if err := sendCommand(mntCmd, "http://unix/mount", ns.MountClient); err != nil {
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:98]: why is `https://` ?

File: pkg/obs/nodeserver.go (hunk 1 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:58]: done

File: pkg/obs/nodeserver.go (hunk 2 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:94]: it will be modified in another PR.

File: pkg/obs/nodeserver.go (hunk 4 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:140]: done

File: pkg/obs/nodeserver.go (hunk 3 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:111]: this passfile will be deleted by hosting environment,
this file would create every time.

File: pkg/obs/mount.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	}
+	defer response.Body.Close()
+	respBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return status.Errorf(codes.Internal, "Failed to read responseBody, err: %v", err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return status.Errorf(codes.Internal, "Failed to execute the command, body: %v", string(respBody))
+	}
```
Comment (chengxiangdong) [inline pkg/obs/mount.go:26]:
```suggestion
	if err != nil {
		return status.Errorf(codes.Internal, "Failed to read responseBody, err: %v", err)
	}
	
	if response.StatusCode != http.StatusOK {
	        respBody, err := ioutil.ReadAll(response.Body)
		return status.Errorf(codes.Internal, "Failed to execute the command, body: %v", string(respBody))
	}
```

**Review by chengxiangdong — COMMENTED** [review]
there is a nit, it will be better after optimization.

File: pkg/obs/mount.go (hunk 6 above)
Comment (deer-hang) [reply pkg/obs/mount.go]: done

**Pull request merged**